rust-symcrypt/src/gcm.rs

```diff
@@ -134,121 +88,22 @@ impl GcmState {
             }
         }
     }
-}
-
-impl Drop for GcmState {
-    fn drop(&mut self) {
-        unsafe {
-            // SAFETY: FFI calls
-            symcrypt_sys::SymCryptWipe(
-                ptr::addr_of_mut!(self.inner.state) as *mut c_void,
-                mem::size_of_val(&mut self.inner.state) as symcrypt_sys::SIZE_T,
-            );
-            symcrypt_sys::SymCryptWipe(
-                ptr::addr_of_mut!(self.inner.expanded_key) as *mut c_void,
-                mem::size_of_val(&mut self.inner.expanded_key) as symcrypt_sys::SIZE_T,
-            );
-        }
-    }
-}
 
-pub fn validate_gcm_parameters(
-    cipher: BlockCipherType,
-    nonce: &[u8; 12], // GCM nonce length must be 12 bytes
-    auth_data: &[u8],
-    data: &[u8],
-    tag: &[u8],
-) -> Result<(), SymCryptError> {
-    unsafe {
-        // SAFETY: FFI calls
-        match symcrypt_sys::SymCryptGcmValidateParameters(
-            convert_cipher(cipher),
-            nonce.len() as symcrypt_sys::SIZE_T,
-            auth_data.len() as symcrypt_sys::UINT64,
-            data.len() as symcrypt_sys::SIZE_T,
-            tag.len() as symcrypt_sys::SIZE_T,
-        ) {
-            symcrypt_sys::SYMCRYPT_ERROR_SYMCRYPT_NO_ERROR => Ok(()),
-            err => Err(err.into()),
-        }
+    pub fn key_len(&self) -> usize {
+        self.0.cbKey as usize
```

Comment:
>self.0.cbKey as usize [](http://example.com/codeflow?start=8&length=21)

You might want to remember the
pub fn new(key: &[u8],

key.length instead of accessing an internal struct's field.